backend/src/controllers/OrderController.ts

```diff
@@ -0,0 +1,158 @@
+    const restaurant = await Restaurant.findById(checkoutSessionRequest.restaurantId);
+
+    if (!restaurant) {
+      throw new AppError('Restaurant not found', 404);
+    }
+
+    const newOrder = new Order({
+      restaurant: restaurant._id,
+      user: req.userId,
+      status: 'placed',
+      cartItems: checkoutSessionRequest.cartItems,
+      deliveryDetails: checkoutSessionRequest.deliveryDetails,
+      createdAt: new Date(),
+    });
+
+    const lineItems = await createLineItems(checkoutSessionRequest, restaurant.menuItems);
+    const session = await createSession(
+      lineItems,
+      newOrder._id.toString(),
+      restaurant.deliveryPrice,
+      restaurant._id.toString(),
+    );
+
+    if (!session.url) {
+      throw new AppError('Error creating checkout session', 500);
+    }
+
+    await newOrder.save();
+    logger.info(`Checkout session created for order ${newOrder._id}`);
+    return res.status(200).json({ url: session.url });
+  } catch (error) {
+    logger.error(
+      `Error creating checkout session: ${error instanceof Error ? error.message : 'Unknown error'}`,
+      { stack: error instanceof Error ? error.stack : undefined },
+    );
+    next(error);
+  }
+};
+
+function createLineItems(
+  checkoutSessionRequest: CheckoutSessionRequest,
+  menuItems: MenuItemType[],
+) {
+  const lineItems = checkoutSessionRequest.cartItems.map((cartItem) => {
+    const menuItem = menuItems.find((item) => item._id.toString() === cartItem.menuItemId);
+    if (!menuItem) {
+      throw new AppError('Menu item not found', 404);
+    }
+
+    const lineItem: Stripe.Checkout.SessionCreateParams.LineItem = {
+      price_data: {
+        currency: 'usd',
+        unit_amount: menuItem.price,
+        product_data: {
+          name: menuItem.name,
+        },
+      },
+      quantity: cartItem.quantity,
+    };
+
+    return lineItem;
+  });
+
+  return lineItems;
+}
+
+async function createSession(
+  lineItems: Stripe.Checkout.SessionCreateParams.LineItem[],
+  orderId: string,
+  deliveryPrice: number,
+  restaurantId: string,
+) {
+  const sessionData = await STRIPE.checkout.sessions.create({
+    line_items: lineItems,
+    mode: 'payment',
+    shipping_options: [
+      {
+        shipping_rate_data: {
+          display_name: 'Delivery',
+          type: 'fixed_amount',
+          fixed_amount: {
+            amount: deliveryPrice,
+            currency: 'usd',
+          },
+        },
+      },
+    ],
+    metadata: {
+      orderId: orderId,
+      restaurantId: restaurantId,
+    },
+    success_url: `${FRONTEND_URL}/order-status?success=true`,
+    cancel_url: `${FRONTEND_URL}/detail/${restaurantId}?cancelled=true`,
+  });
+
+  return sessionData;
+}
+
+const stripeWebhookHandler = async (req: Request, res: Response, next: NextFunction) => {
+  let event;
+  try {
+    const sig = req.headers['stripe-signature'] as string;
+    event = STRIPE.webhooks.constructEvent(req.body, sig, STRIPE_WEBHOOK_SECRET);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/47#issuecomment-2430019945>#d33549</a><span> - 10/22/2024, 07:15 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> It appears that the `STRIPE_WEBHOOK_SECRET` is being used without first checking if it's defined. Consider adding a check to ensure it's not `undefined` before using it.
</div>
<div id="fix">
<b>Fix:</b> Add a check to ensure `STRIPE_WEBHOOK_SECRET` is defined before using it. If it's not defined, throw an error or handle it appropriately.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -129,7 +129,11 @@
    let event;
    try {
      const sig = req.headers['stripe-signature'] as string;
 -    event = STRIPE.webhooks.constructEvent(req.body, sig, STRIPE_WEBHOOK_SECRET);
 +    if (!STRIPE_WEBHOOK_SECRET) {
 +      throw new Error('STRIPE_WEBHOOK_SECRET is not defined');
 +    }
 +    event = STRIPE.webhooks.constructEvent(req.body, sig, STRIPE_WEBHOOK_SECRET);
 +
      if (event.type === 'checkout.session.completed') {
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged